### DISCUSSION TITLE
Porting S2 to qnx700

### DISCUSSION THREAD

**Discussion by BeyondWang** [google/s2geometry#261] [Q&A] [answered]
Hi Guys, 
We team want to use S2 on QNX700, after a quick start through cross-compile, lots of error occurs. 
I do not get any description of QNX. Is it feasible to port it to QNX? Any pages we can refer to?

BRs
Bingyang

**Comment by jmr** [comment]
It should be feasible to port. Only C++11 is required. Do you have abseil-cpp working and the tests passing? What are your errors?

I'm not sure what it takes to get cmake to work with a cross-compiler. Is it finding the QNX include/lib dirs?

**Reply by BeyondWang** [reply]
Hi,

It reports "#error "_Unsupported bytesex: Either IS_BIG_ENDIAN or IS_LITTLE_ENDIAN must be defined"_", after I add "_add_definitions(-DIS_LITTLE_ENDIAN)_" in the toolchain file, it build successfully. I will test if all the functions work as expected.

Thanks a lot.

**Comment by jmr** [comment, marked as answer]
You probably want to add `defined(__QNX__)` (or whatever the appropriate macro is) here:

https://github.com/google/s2geometry/blob/7f0dc0bb9860ecba8ee4b71f93f65d03fb747ec9/src/s2/base/port.h#L151

**Reply by BeyondWang** [reply]
Yes, that's where the error message locates.
